MIGRATION.md

```diff
@@ -0,0 +1,264 @@
+
+- `Utils.randomEntropy` → `randomEntropy`
+- `Utils.randomSecret` → `randomSecret`
+- `Utils.entropyToSecret` → `entropyToSecret`
+- `Utils.secretToEntropy` → `secretToEntropy`
+- `Utils.calculateChecksum` → `calculateChecksum`
+- `Utils.checkChecksum` → `checkChecksum`
+- `Utils.parseSecretString` → `parseSecretString`
+
+### 6. Support for Node 14 has been dropped
+
+Node 14 has stopped receiving security updates since April 2023, and so we’ve decided to no longer support it going forward. Please update to one of the supported versions of Node as listed in xrpl.js’s `README.md`.
+
+### 7. Configuring proxies with the Client
+
+The way to configure proxies for `Client` has changed. It is now done by specifying the `agent` parameter on the `ConnectionOptions` config.
+
+You can generate an `agent` with libraries such as `https-proxy-agent` or any that implements `http.Agent`.
+
+This was done to remove a hard dependency on `https-proxy-agent` when running in the browser and to support `https-proxy-agent@7` which changed several option names. Proxy support was never supported in the browser, and merely burdened xrpl bundles with unused dependencies.
+
+**Before**
+
+`{
+  proxy: `ws://127.0.0.1:${port}`,
+  authorization: 'authorization',
+  trustedCertificates: ['path/to/pem'],
+}`
+
+**After**
+
+ `{
+   agent: new HttpsProxyAgent<string>(`ws://127.0.0.1:${port}`, {
+     ca: ['path/to/pem'],
+   }),
+   authorization: 'authorization'
+ }`
+
+### 8. Bug fix: Setting an explicit `algorithm` when generating a wallet works now
+
+`Wallet.generate()` and `Wallet.fromSeed` were ignoring the `algorithm` parameter. This means that if you were manually specifying `algorithm` in any `Wallet` constructors, you may generate a different `Wallet` keypair when upgrading to 3.0. In that case to get the same generated wallets as before, don’t specify the `algorithm` parameter.
+
+### 9. `AssertionError` → `Error`
+
+In order to get rid of the `assert` polyfill, we’ve replaced `AssertionError`s with `Error` exceptions. We’ve also updated the error messages to be more descriptive. If you were catching those specific errors, you will have to update your catch statements.
+
+This impacts most of `ripple-keypairs` functions but only if you already had issues with incompatible values.
+
+### 10. Pre-bundle browser builds
+
+If you use the pre bundled version of the library you will need to make the following changes:
+
+- Change any references to `dist/browerified.js` to `build/xrplf-secret-numbers-latest.js`.
+- Access any methods as properties of `xrplf_secret_numbers` instead of using browserify's loader.
+
+### 11. Transaction` type
+
+`Transaction` has been updated to include `PsuedoTransaction`s. To get the equivalent of the old `Transaction` type which only included transactions users could submit, please use `SubmittableTransaction`.
```

Comment:
typo in `PseudoTransaction`. The `*.ts` code files have the correct spelling of the word